crates/kitsune_p2p/kitsune_p2p/src/spawn/actor/space.rs

```diff
@@ -631,6 +652,72 @@ impl Space {
         }
     }
 
+    fn publish_leave_agent_info(
+        &mut self,
+        agent: Arc<KitsuneAgent>,
+    ) -> KitsuneP2pHandlerResult<()> {
+        let space = self.space.clone();
+        let network_type = self.config.network_type.clone();
+        let evt_sender = self.evt_sender.clone();
+        let bootstrap_service = self.config.bootstrap_service.clone();
+        let expires_after = self.config.tuning_params.agent_info_expires_after_ms as u64;
+        Ok(async move {
+            let signed_at_ms = crate::spawn::actor::bootstrap::now_once(None).await?;
+            let expires_at_ms = signed_at_ms + expires_after;
+            use kitsune_p2p_types::agent_info::AgentInfoSigned;
+            let agent_info_signed = AgentInfoSigned::sign(
+                space.clone(),
+                agent.clone(),
+                0,          // no storage arc
+                Vec::new(), // no urls
```

Comment:
urls empty here which will hit the above error?

Reply:
Yep, this is needed to propagate the "inactive" status of this node around the network... I've attempted to filter these out everywhere we might be searching for an active node to talk to, but not from the places where we gossip these entries (because otherwise the "active" versions would keep gossiping).

Hopefully I got all this logic right : )